File: python/7 kyu/199.py
# https://www.codewars.com/kata/59778cb1b061e877c50000cc/train/python

# Passed

def arr_adder(arr):
    words = [str()] * len(arr[0])
    for row in arr:
        for idx, char in enumerate(row):
            words[idx] += char
            
    sentence = " ".join(words)
    return sentence

output = arr_adder([
    ['J','L','L','M'],
    ['u','i','i','a'],
    ['s','v','f','n'],
    ['t','e','e','' ]
])
print(output)

File: nodejs/7 kyu/199.js
// https://www.codewars.com/kata/5808c8eff0ed4210de000008/train/javascript

// Passed

function part(x) {
  let alanTerms = [
    "partridge",
    "peartree",
    "chat",
    "dan",
    "toblerone",
    "lynn",
    "alphapapa",
    "nomad",
  ];
  let validTerms = x.filter((term) => alanTerms.includes(term.toLowerCase()));
  let result =
    validTerms.length === 0
      ? "Lynn, I've pierced my foot on a spike!!"
      : "Mine's a Pint" + "!".repeat(validTerms.length);
  return result;
}

const output = part([
  "Grouse",
  "Partridge",
  "Pheasant",
  "Goose",
  "Starling",
  "Robin",
  "Thrush",
  "Emu",
  "PearTree",
  "Chat",
  "Dan",
  "Square",
  "Toblerone",
  "Lynn",
  "AlphaPapa",
  "BMW",
  "Graham",
  "Tool",
  "Nomad",
  "Finger",
  "Hamster",
]);
console.log(output);